File: MonitoringAgent/MonitoringAgent.Data.Interfaces/Entities/GetSitesStatus.cs
using System;

namespace MonitoringAgent.Data.Interfaces.Entities
{
    public partial class GetSitesStatus: ISystemFields
    {
		public GetSitesStatus()
		{
			ChangeDate = CreateDate = DateTime.Now;
		}
        public int Id { get; set; } // ID
        public int? CheckStatus { get; set; } // CHECK_STATUS
        public DateTime? CheckDate { get; set; } // CHECK_DATE
        public string Message { get; set; } // MESSAGE
        public int? Attempt { get; set; } // ATTEMPT
        public string Name { get; set; } // NAME
        public string SitePath { get; set; } // SITE_PATH
        public DateTime? DeleteDate { get; set; } // DELETE_DATE
        public DateTime CreateDate { get; set; } // CREATE_DATE
        public DateTime ChangeDate { get; set; } // CHANGE_DATE
        public int? LogTypeInfoId { get; set; } // LOG_TYPE_INFO_ID
	}
}

File: MasterDataModule/MasterDataModuleWeb/web/models/Settings/GetSitesStatus.js
define(function () {
	'use strict';

	var model = Backbone.Model.extend({
	    urlRoot: 'api/GetSitesStatuses',
		fields: {
			id: { type: "number", editable: false }
			,checkStatus: { type: "number", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'checkStatus'), 
				                    validation: { required: false } }			
			,checkDate: { type: "date", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'checkDate'), 
				                    validation: { required: false, date: true } }			
			,message: { type: "string", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'message'), 
				                    validation: { required: false } }			
			,attempt: { type: "number", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'attempt'), 
				                    validation: { required: false } }			
			,name: { type: "string", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'name'), 
				                    validation: { required: true } }			
			,sitePath: { type: "string", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'sitePath'), 
				                    validation: { required: true } }			
			,logTypeInfoId: { type: "number", 
			                        editable: Application.canTableItemBeEdit('GetSitesStatus', 'logTypeInfoId'), 
				                    validation: { required: true } }			
		},
		defaults: function () {
			var dnf = new Date();
			var dnt = new Date(2070,11,31);
			return {
				fromDate: dnf, 
				toDate: dnt
			};
		}
	});
	return model;
});
